src/conv.rs

```diff
@@ -207,7 +207,17 @@ pub fn map_device_descriptor<'a>(
                 label: OwnedLabel::new(extras.label).into_cow(),
                 features: features,
                 limits: wgt::Limits {
-                    max_bind_groups: extras.maxBindGroups,
+                    max_texture_dimension_1d: extras.maxTextureDimension1D | 8192,
```

Comment:
this is scary!
Why are we using binary OR for them?

Reply:
If by "we" you mean *me*, I'm just using patterns that I know :)  - is this style frowned upon in Rust? How would I do it instead?

Reply:
"we" is a polite term here for "you the author plus us in a virtual term". I.e. if I'm trying to say something is wrong, shifting the responsibility from "you" to "we" makes it much easier for you as the author accept it :)

> is this style frowned upon in Rust? How would I do it instead?


It's not exactly about style. Rather, it's not going to handle cases where `maxTextureDimension1D` is not zero. Say, it's 16384. By `|` you'd produce a value `16384 | 8192` which is bigger than what user requested, so it's unexpected.

There are many ways to approach this, but generally it's hard to be DRY.
The simplest correct way would be:
```
max_texture_dimension_1d: if extras.maxTextureDimension1D != 0 { extras.maxTextureDimension1D } else { defaults.max_texture_dimension_1d },
```

Reply:
Thanks for the explanation. I thought you may have thought I copied that code from elsewhere in the codebase or something. But not being a native English may have contributed to the confusion :)

Oh boy, I meant using a logical or (shortcircuit), not a bitwise or.

Reply:
ah, I see. Right, C/C++ has this pattern in places. I didn't see it used in Rust.

Reply:
also, not a native speaker, so take my advice with a grain of salt :)

Reply:
But of course a logical or won't work since Rust won't implicitly cast to a bool :)

Reply:
Updated.